src/stratigraphy/util/geometric_line_utilities.py

```diff
@@ -223,36 +229,106 @@ def merge_parallel_lines(lines: list[Line], tol: int = 8, angle_threshold: float
+        line2 (Line): Second line to merge.
 
     Returns:
         Line: The merged line.
     """
-    # determine start point
-    if line1.start.x > line2.start.x:
-        x_start = line2.start.x
-        y_start = line2.start.y
-    else:
-        x_start = line1.start.x
-        y_start = line1.start.y
+    x = np.array([line1.start.x, line1.end.x, line2.start.x, line2.end.x])
+    y = np.array([line1.start.y, line1.end.y, line2.start.y, line2.end.y])
+    slope, intercept = odr_regression(x, y)
+    if np.isnan(slope):
+        logger.warning("Merged line slope is nan. Returning line1 in absence of any better choice.")
+        return line1
+    projected_points = []
+    for point in [line1.start, line1.end, line2.start, line2.end]:
+        projection = _get_orthogonal_projection_to_line(point, slope, intercept)
+        projected_points.append(projection)
+    # Take the two points such that the distance between them is the largest.
+    # Since there are only 4 points involved, we can just calculate the distance between all points.
+    point_combinations = list(combinations(projected_points, 2))
+    distances = []
+    for interval in point_combinations:
+        distances.append(_calculate_squared_distance_between_two_points(interval[0], interval[1]))
+
+    point1, point2 = point_combinations[np.argmax(distances)]
+    return Line(point1, point2)
+
+
+def _calculate_squared_distance_between_two_points(point1: Point, point2: Point) -> float:
+    return (point1.x - point2.x) ** 2 + (point1.y - point2.y) ** 2
 
-    # determine end point
-    if line1.end.x > line2.end.x:
-        x_end = line1.end.x
-        y_end = line1.end.y
-    else:
-        x_end = line2.end.x
-        y_end = line2.end.y
 
-    return Line(Point(x_start, y_start), Point(x_end, y_end))
+def _get_orthogonal_projection_to_line(point: Point, slope: float, intercept: float) -> Point:
+    """Calculate orthogonal projection of a point onto a line in 2D space.
+
+    Formula taken from:
+    https://math.stackexchange.com/questions/62633/orthogonal-projection-of-a-point-onto-a-line
+
+    Args:
+        point (Point): The point to project onto the line.
+        slope (float): Slope of the line.
+        intercept (float): Intercept of the line.
+
+    Returns:
+        Point: Projected point onto the line.
+    """
+    v = np.array([1, slope])
+    projection = np.dot(np.outer(v, v) / np.inner(v, v), np.array([point.x, point.y - intercept])) + np.array(
+        [0, intercept]
+    )
+    if np.shape(projection) != (2,):
+        logger.critical(
+            f"Wrong shape for orthogonal projection. The correct shape is (2,), but it is {np.shape(projection)}."
+        )
+        raise ValueError
+    if np.isnan(projection).any():
+        logger.critical("Projection is nan. Cannot calculate orthogonal projection.")
+        raise ValueError
+    return Point(int(np.round(projection[0], 0)), int(np.round(projection[1], 0)))
```

Comment:
It's because we operate on a pixel grid, and some downstream functionalities such as plotting might not work any more. 

I even thought about forcing the class object Point to take integers only.

I believe plotting with cv2 is the only thing that doesnt work. (Mainly as we don't use any other cv2 functionalities afterwards any more).